repository/kafka/operator/templates/bootstrap.yaml

```
@@ -97,3 +97,25 @@ data:
     # Provided configuration
     ${KAFKA_CONFIGURATION}
     EOF
+
+    {{ if .Params.CUSTOM_SERVER_PROPERTIES_CM_NAME }}
+    KAFKA_CUSTOM_CONFIGURATION_FILE=$(ls -d /custom-configuration/* | head -n 1)
+    cp $KAFKA_CUSTOM_CONFIGURATION_FILE /tmp/custom-configuration
+    echo "Appending custom configuration file to the server.properties..." | xargs -L 1 echo $(date +'[%Y-%m-%d %H:%M:%S,%3N]') $1
+
+    # remove blacklisted properties that users cannot override
+    skip_properties=("broker.id" "listeners" "advertised.listeners" "advertised.host.name" "listener.security.protocol.map" "log.dirs")
+    for property in "${skip_properties[@]}"; do
+    	if grep -q $property /tmp/custom-configuration; then
+      		echo "WARN: cannot override the $property using custom properties configmap.";
+      		echo "Removing property $property"
+      		sed -i "/^$property/d" /tmp/custom-configuration
+    	fi
+    done
+
+    KAFKA_CUSTOM_CONFIGURATION=$(cat /tmp/custom-configuration)
+    cat /tmp/custom-configuration
```

```suggestion
    cat ${KAFKA_CUSTOM_CONFIGURATION}
```
might as well use it if you saved it to a variable :)

---

cannot cat a cat, but yeah we can fix it another way

---

fixed this!

---

oh hahaha `cat a cat` sorry, terrible suggestion on my part